src/cpp/src/tokenizer.cpp

```diff
@@ -0,0 +1,206 @@
+}
+
+}
+
+namespace ov {
+namespace genai {
+
+class Tokenizer::TokenizerImpl {
+public:
+    ov::InferRequest m_tokenize_request;
+    ov::InferRequest m_detokenizer_request;
+    int64_t m_pad_token_id = 0;
+    int64_t m_bos_token_id = 1;
+    int64_t m_eos_token_id = 2;
+
+    TokenizerImpl() = default;
+    TokenizerImpl(std::string tokenizers_path, const std::string device, const std::string& ov_tokenizers_path) {
+        ov::Core core;
+        
+        if (ov::genai::utils::is_xml(tokenizers_path))
+            OPENVINO_THROW("tokenizers_path should be a path to a dir not a xml file");
+    
+        if (ov_tokenizers_path.empty()) {
+            // OPENVINO_TOKENIZERS_PATH is defined in CMakeLists.txt
+            core.add_extension(OPENVINO_TOKENIZERS_PATH);
+        } else {
+            core.add_extension(ov_tokenizers_path + "/libopenvino_tokenizers.so");
+        }
+        std::shared_ptr<ov::Model> tokenizer_model, detokenizer_model;
+        try {
+            tokenizer_model = core.read_model(tokenizers_path + "/openvino_tokenizer.xml");
+            detokenizer_model = core.read_model(tokenizers_path + "/openvino_detokenizer.xml");
+        } catch (...) {
+            OPENVINO_THROW("Cannot compile tokenizer and/or detokenizer. Please check that "
+                        "openvino_tokenizer.xml and openvino_detokenizer.xml exist in \"" + tokenizers_path + "\"");
+        }
+        m_tokenize_request = core.compile_model(tokenizer_model, device).create_infer_request();
+        m_detokenizer_request = core.compile_model(detokenizer_model, device).create_infer_request();
+
+        auto rt_info = tokenizer_model->get_rt_info();
+        if (rt_info.count("eos_token_id") > 0)
+            m_eos_token_id = rt_info["eos_token_id"].as<int64_t>();
+        if (rt_info.count("bos_token_id") > 0)
+            m_bos_token_id = rt_info["bos_token_id"].as<int64_t>();
+        if (rt_info.count("pad_token_id") > 0)
+            m_pad_token_id = rt_info["pad_token_id"].as<int64_t>();
+        }
+
+    std::pair<ov::Tensor, ov::Tensor> encode(std::string prompt) {
+        size_t batch_size = 1;
+        m_tokenize_request.set_input_tensor(ov::Tensor{ov::element::string, {batch_size}, &prompt});
+        m_tokenize_request.infer();
+        return {m_tokenize_request.get_tensor("input_ids"), m_tokenize_request.get_tensor("attention_mask")};
+    }
+
+    std::pair<ov::Tensor, ov::Tensor> encode(std::vector<std::string>& prompts) {
+        m_tokenize_request.set_input_tensor(ov::Tensor{ov::element::string, {prompts.size()}, prompts.data()});
+        auto size_ = m_tokenize_request.get_input_tensor().get_shape();
+        m_tokenize_request.infer();
+        pad_left(m_tokenize_request.get_tensor("input_ids"), m_tokenize_request.get_tensor("attention_mask"), m_pad_token_id);
+        
+        // todo: fix mask filled with '2' instead of '0' 
+        // https://github.com/openvinotoolkit/openvino_tokenizers/pull/90 should've fixed this
```

Comment:
it's fixed already (https://github.com/openvinotoolkit/openvino_tokenizers/pull/90 is merged), can we drop WA?

Reply:
unfortunately, error is still present and without w/a does not work